contracts/serverless-v2/CommonChainExecutors.sol

```diff
@@ -0,0 +1,305 @@
+
+contract CommonChainExecutors is
+    Initializable, // initializer
+    ContextUpgradeable, // _msgSender, _msgData
+    ERC165Upgradeable, // supportsInterface
+    AccessControlEnumerableUpgradeable, // RBAC enumeration
+    AttestationAutherUpgradeable,
+    TreeUpgradeable,
+    UUPSUpgradeable // public upgrade
+{
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    // initializes the logic contract without any admins
+    // safeguard against takeover of the logic contract
+    constructor(
+        IAttestationVerifier attestationVerifier,
+        uint256 maxAge
+    ) AttestationAutherUpgradeable(attestationVerifier, maxAge) initializer {}
+
+    using SafeERC20 for IERC20;
+    using ECDSA for bytes32;
+
+    modifier onlyAdmin() {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()), "only admin");
+        _;
+    }
+
+    modifier onlyJobsContract() {
+        require(_msgSender() == address(jobs), "ONLY_JOBS_CONTRACT");
+        _;
+    }
+
+    //-------------------------------- Overrides start --------------------------------//
+
+    function supportsInterface(
+        bytes4 interfaceId
+    )
+        public
+        view
+        virtual
+        override(ERC165Upgradeable, AccessControlEnumerableUpgradeable)
+        returns (bool)
+    {
+        return super.supportsInterface(interfaceId);
+    }
+
+    function _grantRole(bytes32 role, address account) internal virtual override(AccessControlEnumerableUpgradeable) returns (bool) {
+        return super._grantRole(role, account);
+    }
+
+    function _revokeRole(bytes32 role, address account) internal virtual override(AccessControlEnumerableUpgradeable) returns (bool) {
+        bool res = super._revokeRole(role, account);
+
+        // protect against accidentally removing all admins
+        require(getRoleMemberCount(DEFAULT_ADMIN_ROLE) != 0, "AV:RR-All admins cant be removed");
+        return res;
+    }
+
+    function _authorizeUpgrade(
+        address /*account*/
+    ) internal view override onlyAdmin {}
+
+    //-------------------------------- Overrides end --------------------------------//
+
+    //-------------------------------- Initializer start --------------------------------//
+
+    function __CommonChainExecutors_init(
+        address _admin,
+        EnclaveImage[] memory _images,
+        IERC20 _token
+    ) public initializer {
+        require(_admin != address(0), "ZERO_ADDRESS_ADMIN");
+        require(address(_token) != address(0), "ZERO_ADDRESS_TOKEN");
+
+        __Context_init();
+        __ERC165_init();
+        __AccessControlEnumerable_init();
+        __AttestationAuther_init_unchained(_images);
+        __TreeUpgradeable_init_unchained();
+        __UUPSUpgradeable_init();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+
+        token = _token;
+    }
+
+    //-------------------------------- Initializer end --------------------------------//
+
+    IERC20 public token;
+    CommonChainJobs public jobs;
```

Comment:
can remove, unused.

Reply:
it is required for onlyJobsContract() modifier

Reply:
Okay